---
title: "20220609_proteomic_landscape_npm1_coregulation_network"
output: github_document
---

# Introduction

Proteomics, as opposed to genomics, is much better suited to create networks of protein-protein interaction, co-occurance and co-regulation. While for protein-protein interactions, targeted assays such as BioID are superior, broad patterns of co-regulations can be readily build from global proteomic data. 

Kustatscher at al. have outlined a computational framework for the creation of a protein co-regulation map at https://www.nature.com/articles/s41587-019-0298-5

The creation can be split into two sub-processes, one is the creation of an suitable metric to assess the degree of co-regulation/correlation and the second is the modelling of the underlying network based on that metric.

Kustatscher et al. propose using the treeClust algorithm to model the co-regulation as opposed to traditionally used methods such as pearson correlation coefficient (PCC). They show in a set of simulation experiments, that treeClust, enhanced with an additional topological overlap measure (TOM), outperforms PCC (much lower fraction of false-positives).

They tuned a hyperparameter optimazation for treeClusts serule (tree pruning) and complexity using a grid search against a Reactome gold standard of protein-protein interactions. The optimal hyperparameters were found the be serule = 1.8 and cp = 0.105

Code is found at 

https://github.com/Rappsilber-Laboratory/ProteomeHD/blob/master/R%20scripts/tSNE_map.R#L11

## Building a treeClus dis-similarity matrix

```{r}
require(treeClust)
require(WGCNA)

#calculate the treeClust distance using the hyperparameters from Kustatscher et al.

tc_distances <- treeClust::treeClust.dist(as.data.frame(prot_merge),
                               d.num = 2,
                               verbose = TRUE,
                               rcontrol = rpart.control(cp = 0.105),
                               control = treeClust.control(serule = 1.8) )

## Turn the distance matrix into a similarity matrix
tc_sim_symm <- 1-as.matrix(tc_distances)
```

## Constructing the WGCNA adjacency matrix

After creating the similarity matrix, we use WGCNA sigmoid function to build an adjacency matrix. Here again, the two hyperparameters were tuned based on the Reactome gold standard

```{r}
adj_mat <- WGCNA::sigmoidAdjacencyFunction(tc_sim_symm, mu = 0.91, alpha = 37)
```

Finally, WGCNA::TOM is used to re-weight the matrix

```{r}
adj_mat_tom <- WGCNA::TOMsimilarity( adj_mat, TOMDenom = "mean" )
colnames(adj_mat_tom) <- colnames(adj_mat)
rownames(adj_mat_tom) <- colnames(adj_mat)
```

# Visualization

We can use UMAP to visualize the protein-protein distances (1- similarity). UMAP has several hyperparameters which have significant impact on the final representation. Details can be found here

https://umap-learn.readthedocs.io/en/latest/parameters.html



```{r}
#library(Rtsne)

1-log2(adj_mat_tom) -> adj_mat_tom.dist

umap::umap(adj_mat_tom.dist, 
           input = "dist", 
           verbose = T, 
           spread = 5, 
           min_dist = 0.1, 
           n_neighbors = 10, 
           n_components = 2) -> umap.out

#Rtsne::Rtsne(adj_mat_tom.dist, theta = 0, perplexity = 30, max_iter = 1000, is_distance = T) -> t_sne.out
```


### UMAP

Clusters are identified using HDBSCAN. eps influences the number of clusters. For a better separation, one could for example increase the number of UMAP components and use the larger space for clustering. 

Finally, ggrepel is used to annotate the cluster numbers.

```{r fig.width=8, fig.height=6}

dbscan::dbscan(umap.out$layout, eps = 0.6) -> dbscan.out

umap.out$layout %>% 
  as_tibble(rownames = "uniprot") %>% 
  bind_cols(cluster = dbscan.out$cluster) %>% 
  group_by(cluster) %>% 
  summarize(V1 = median(V1), V2 = median(V2)) -> ggrepel_df

umap.out$layout %>% 
  as_tibble(rownames = "uniprot") %>% 
  bind_cols(cluster = dbscan.out$cluster) %>% 
  ggplot(aes(x=V1, y = V2, col = factor(cluster), label = cluster)) +
  geom_point() +
  cowplot::theme_cowplot() +
  ggrepel::geom_label_repel(data = ggrepel_df, aes(label = cluster))

ggsave("npm1_proteome_wgcna_umap.png", device = "png", dpi = 300, width = 8, height = 6)
```


## tSNE

An alternative to UMAP is tSNE. Here, we need to tune two hyperparameters, the number of PCs to use and the perplexity. 

There's a extensive towardsdatascience article by Nikolay Oskolkov that discusses quasi-empirical ways to approach this

https://towardsdatascience.com/how-to-tune-hyperparameters-of-tsne-7c0596a18868

i.e. using an initial estimate for perplexity as `r ncol(adj_mat)^(1/2)` and estimating the number of informative PCs via permutation

```{r}
prcomp(adj_mat_tom.dist)
```



## GO over representation analyses

We can run a simple enrichment cluster-wise enrichment analysis by pulling the protein IDs of the cluster members and comparin against the universe of all proteins.

```{r}
umap.out$layout %>% 
  as_tibble(rownames = "uniprot") %>% 
  bind_cols(cluster = dbscan.out$cluster) %>% 
  filter(cluster == 7) %>% 
  pull(uniprot) %>% 
  clusterProfiler::enrichGO(gene = ., 
                            OrgDb = "org.Hs.eg.db", 
                            ont = "CC", 
                            pvalueCutoff = 0.05, 
                            minGSSize = 5, 
                            keyTypSe = "UNIPROT", 
                            universe = as.character(rownames(adj_mat_tom))) %>% 
  as_tibble()
```

